plugin/hash.lua

```diff
@@ -1,83 +1,67 @@
-				.. "be able to send messages to it")
+		beerchat.set_player_channel(name, channel)
+		if channel == beerchat.main_channel_name then
+			minetest.chat_send_player(
+				name,
+				"Switched to channel " .. channel .. ", messages will now be sent to this channel"
+			)
 		else
-			if channel_name == "" then--use last used channel
-				-- We need to get the target
-				channel_name = hashchat_lastrecv[name]
-			end
-			if channel_name and channel_name ~= "" then
-				beerchat.on_channel_message(channel_name, name, msg)
-				beerchat.send_on_channel(name, channel_name, msg)
-			else
-				return false
-			end
+			minetest.chat_send_player(
+				name,
+				"Switched to channel " .. channel .. ", messages will now be sent to this channel. "
+				.. "To switch back to the main channel, type #" .. beerchat.main_channel_name
+			)
+		end
+		if beerchat.enable_sounds then
+			minetest.sound_play(
+				beerchat.channel_management_sound,
+				{
+					to_player = name,
+					gain = beerchat.sounds_default_gain
+				}
+			)
 		end
+	end
+end
+
+beerchat.register_on_chat_message(function(name, message)
+	local channel_name, msg = string.match(message, "^#(%S+) ?(.*)")
+
+	if not channel_name then
+		return false
+	elseif msg == "" then
+		switch_channel(name, channel_name)
 		return true
-	else
-		channel_name = string.match(message, "^#(.*)")
-		if channel_name then
-			if not beerchat.channels[channel_name] then
-				minetest.chat_send_player(name, "Channel " .. channel_name
-					.. " does not exist")
-			elseif not beerchat.is_player_subscribed_to_channel(name, channel_name) then
-				minetest.chat_send_player(name, "You need to join this channel in order "
-					.. "to be able to switch to it")
-			else
-				if not beerchat.execute_callbacks('before_switch_chan', name,
-					beerchat.currentPlayerChannel[name], channel_name) then
-					return false
-				end
-				beerchat.set_player_channel(name, channel_name)
-				if channel_name == beerchat.main_channel_name then
-					minetest.chat_send_player(
-						name,
-						"Switched to channel " .. channel_name
-						.. ", messages will now be sent to this channel"
-					)
-				else
-					minetest.chat_send_player(
-						name,
-						"Switched to channel " .. channel_name
-						.. ", messages will now be sent to this channel. To switch back "
-						.. "to the main channel, type #" .. beerchat.main_channel_name
-					)
-				end
+	end
 
-				if beerchat.enable_sounds then
-					minetest.sound_play(beerchat.channel_management_sound, {
-						to_player = name, gain = beerchat.sounds_default_gain })
-				end
-			end
-			return true
-		end
+	if not beerchat.execute_callbacks('before_send', name, msg, channel_name) then
 		return false
 	end
+
+	if not beerchat.channels[channel_name] then
+		minetest.chat_send_player(name,
+			"Channel " .. channel_name .. " does not exist. Make sure the channel still exists "
+			.. "and you format its name properly, e.g. #channel message or #my channel: message"
```

Comment:
Help texts still contain references to `#my channel: message` format, these should also be removed.

Reply:
10b7e98 ... and of course couldn't leave it at that, few simple dry changes